test/libyul/Parser.cpp

```diff
@@ -563,6 +565,47 @@ BOOST_AUTO_TEST_CASE(builtins_analysis)
 	CHECK_ERROR_DIALECT("{ let a, b := builtin(1, 2) }", DeclarationError, "Variable count mismatch: 2 variables and 3 values.", dialect);
 }
 
+BOOST_AUTO_TEST_CASE(default_types_set)
+{
+	ErrorList errorList;
+	ErrorReporter reporter(errorList);
+	shared_ptr<Block> result = parse(
+		"{"
+			"let x:bool := true:bool "
```

Comment:
Perhaps not in the scope of this PR: is `true:bool` needed and planned to be deducted in the future? If a boolean value is greater than 1, would that mean that `0:bool` and `1:bool` are valid, whereas e.g. `2:bool` wouldn't?

Reply:
I don't think `0:bool` or `1:bool` should be allowed.

Reply:
isn't `true` intrinsically boolean already?

Reply:
I think there is a test that shows that you can omit the `:bool` for `true`.